# Völuspá's vocabulary, orthographic rules and  annotations

The *Heimskringla* project <a id="heimskringla"></a> is an independent project

In [1]:
from zoegas import reader
from eddas import reader as poetic_edda_reader
from eddas import pos as poetic_edda_pos

We use the digitalized Zoëga's dictionary or *ZD* (**zoegas**) and the digitalized Völuspá (**eddas**).

In [2]:
from collections import defaultdict

In [3]:
PUNCTUATIONS = "':?!.,;-"

In [4]:
dictionary = reader.Dictionary(reader.dictionary_name)

In [5]:
word = dictionary.find("heimr")

In [6]:
lem_voeluspaa = poetic_edda_reader.PoeticEddaLemmatizationReader("Völuspá")

In [7]:
tagged_words = lem_voeluspaa.tagged_words()

In [8]:
tagged_words

[('1', ''), ('Hljóðs', 'HLJÓÐ'), ('bið', 'BIÐJA'), ...]

In [9]:
dictionary_entries = {}
forms = defaultdict(list)
proper_nouns = []
for word, tag in tagged_words:
    if tag.lower() not in dictionary_entries and tag != '' and tag not in PUNCTUATIONS:
        dictionary_entries[tag.lower()] = dictionary.find(tag.lower())
    if len(word) > 0 and word[0].isupper():
        proper_nouns.append(word)
    forms[tag.lower()].append(word)

In [10]:
# Words not in dictionary
not_found_lemmata = []
for lemma in dictionary_entries:
    if dictionary_entries[lemma] is None:
        not_found_lemmata.append(lemma)

In [11]:
print(" ".join(not_found_lemmata))

heimdallr valföðr fyr öld upphiminn mæran grænn hægr hon iðavöllr gøra gullr uns jötunheimr bláinn móðsognir mættr durinn alþjófr dvalinn nípingr bívurr bávurr bömburr nóri ánn ánarr óinn mjöðvitnir gandalfr vindalfr þorinn þráinn nýráðr reginn ráðsviðr fíli kíli náli hefti víli hannar svíurr billingr brúni buri hornbori lóni aurvangr jari eikinskjaldi kindr lofarr sökja aurvanga jöruvalla draupnir dolgþrasir haugspori hlévangr glóinn dóridóri óri dúfr skirfir virfir skáfiðr alfr yngvi fjalarr frosti finnr ginnarr lofar und+es það öflugr ask embla örlöglauss önd óðinn hænir lóðurr góð yggdrasill ausinn þær+es urðarbrunnr verðandi örlög seggja folkvígr gullveig þrisvar ósjaldan heiður hvar+es velspá skylu folk vanir vígspá þórr slíkt ér mímisbrunnr mjöð mímir morgun hvern herföðr goðþjóð skögul göndul geirskögul herjans baldr blóðigr týr höðr neka frigg fensalir váli vígband hveralundr loki sigyn glýja niðavallr sindri ókólnn brimir ljóra niðhöggr járnviðr fenrir trölls eftir eggþér gal

In [12]:
print(" ".join(proper_nouns))

Hljóðs Heimdallar Valföðr Ek Ár Áðr Burs Miðgarð Sól Þá Hittusk Iðavelli Tefldu Jötunheimum Þá Brimis Bláins Þar Móðsognir Durinn Durinn Nýi Niði Norðri Suðri Austri Vestri Alþjófr Dvalinn Nár Náinn Nípingr Dáinn Bívurr Bávurr Bömburr Nóri Ánn Ánarr Óinn Mjöðvitnir Veggr Gandalfr Vindalfr Þorinn Þrár Þráinn Þekkr Litr Vitr Nýr Nýráðr Reginn Ráðsviðr Fíli Kíli Fundinn Náli Hefti Víli Hannar Svíurr Billingr Brúni Bíldr Buri Frár Hornbori Frægr Lóni Aurvangr Jari Eikinskjaldi Mál Dvalins Lofars Aurvanga Jöruvalla Þar Draupnir Dolgþrasir Hár Haugspori Hlévangr Glóinn DóriDóri Óri Dúfr Andvari Skirfir Virfir Skáfiðr Ái Alfr Yngvi Eikinskjaldi Fjalarr Frosti Finnr Ginnarr Lofars Unz Ask Emblu Önd Óðinn Hænir Lóðurr Ask Yggdrasill Urðarbrunni Þaðan Urð Verðandi Skuld Þat Gullveigu Hárs Heiði Þá Fleygði Óðinn Þá Óðs Þórr Veit Heimdallar Valföðrs Vituð Ein Hvers Hví Allt Óðinn Mímisbrunni Drekkr Mímir Valföðrs Vituð Valði Herföðr Sá Goðþjóðar Skuld Skögul Gunnr Hildr Göndul Geirskögul Nú Herjan

In [13]:
len(not_found_lemmata), len(dictionary_entries)

(188, 701)

*ZD* does not contain any proper nouns. 

In [14]:
gods = ['heimdallr', 'valföðr', 'baldr', 'sigföður', 'óðinn', 'yngvi','frigg','þórr', 'loki', 'hænir', 'surtr', 'váli',
       'týr', 'verðandi', 'herföðr', 'víðarr', 'hlóðyn', 'sigyn', 'hlín', 'höðr', 'brimir', 'fjörgyn', 'hveðrungr', 
        'hroptr', 'lóðurr', 'herjan', ]
dwarves = ['gandalfr', 'durinn', 'fíli', 'kíli', 'náli', 'durinn', 'dvalinn', 'alþjófr', 'bívurr', 'bávurr', 'bömburr', 'nóri',
           'ánn', 'ánarr', 'óinn', 'mjöðvitnir', 'gandalfr', 'vindalfr', 'þorinn', 'þráinn', 'nýráðr', 'reginn', 'ráðsviðr', 
           'fíli', 'kíli', 'náli', 'hefti', 'víli', 'hannar', 'svíurr', 'billingr', 'brúni', 'buri', 'hornbori', 'lóni',
           'aurvangr', 'jari', 'eikinskjaldi', 'lofarr', 'dolgþrasir',
           'dóridóri', 'óri','glóinn', 'aurvanga', 'haugspori', 'hlévangr',
           'dúfr', 'skirfir', 'virfir', 'skáfiðr', 'frosti', 'fjalarr', 'sindri',
          ]
creatures = ['alfr', 'ask', 'embla', 'fenrir', 'eggþér', 'trölls', 'galgviðr', 'gullinkambr', 'niðhöggr',]
species_or_groups = ['vanir', 'goðþjóð',]
giants = ['eggþér', 'hrymr', ]
names = ['finnr', 'heiður', ]
places = ['iðavöllr', 'jötunheimr', 'iðavöllr', 'móðsognir', 'múspell', 'fensalir', 'yggdrasill',
          'jöruvalla', 'urðarbrunnr', 'niðavallr', 'gimlé', 'gnipahellir', 'hveralundr', 'járnviðr',] # 'jöruvalla' -> 'jöruvöllr',
unreferenced_words = ['fyr', 'upphiminn', 'hon',]
spelling_variants = {'eftir': 'eptir', 'folk': 'fólk', 'skylu': "skulu", 'loft': 'lopt', 'seggja': 'seggr',}
mistakes = {'bláinn': '', 'kindr': 'kind', 'ulfr': 'úlfr', 'ausinn': 'ausa', 'slíkt': 'slíkr', 'gullr': 'gull',
            'morgun': 'morgunn', 'öflugr': 'öflugr','velspá': 'velspár', 'það': 'þat', 'ókólnn': 'ókólnir', 'neka': 'nema',
            'sigtoftir': 'sigtoptir', 'fjölð': 'fjöld', 'niðavallr': 'niðavöllr', 'flýgja': 'fljúga', 'býleists': 'býleistr',
           'nken': 'föður', 'valtívar': 'valtýr', 'vígspá': 'vígspár', 'iðjagræn': 'iðjagrænn', 'sigtívamr': 'sigtýr', 
            'árdagr': 'árdagar', 'ókvíð': 'ókvíðinn', 'lkfnsf': 'allr', 'ramr': 'rámr', 'sjalfan': 'sjálfr', 'skula': 'skulu',
           'einr': 'eimr', 'ljóra': 'ljóri', 'góð': 'góðr', 'forsi': 'fors', 'þann': 'sá', 'beli': 'belja', 
            'folkvígr': 'fólkvíg', 'mjöð': 'mjöðr', 'þingi': 'þing', 'hroft': 'hroptr', 'blóðigr': 'blóðga', 'ér': 'þú', 
           'herjans': 'herjan', 'ræðr': 'ráða', 
           }
valkyries = ['göndul', 'skögul', 'geirskögul', ]
witches = ['gullveig', ]
objects = ['draupnir', 'naglfar', 'gjallarhorn', ]

merges = ['uns', 'þær+es', 'hvar+es', 'sá+er', 'und+es',]
compounds = ['vígband', 'skalmöld', ]

not_found = ['glýja', 'þrisvar',]
not_parsed_correctly = ['moldþinnur',]
different_orthographic_conventions = {'öld': '', 'mæran': '', 'grænn': '', 'hægr': '', 'gøra': '', 'mættr': '', 'nípingr': '',
 'sökja': '', 'ginnarr': '', 'lofar': '', 'örlöglauss': '', 'önd': '', 'örlög': '', 'ósjaldan': '', 'hvern': '',
 'rökr': '', 'niðfölr': '', 'örn': '', 'dæma': 'dæma', 'fegr': ''}
proper_nouns = gods + dwarves + creatures + species_or_groups + giants + names + places + unreferenced_words +\
    objects + merges + compounds + valkyries + witches 
    
dictionary_errors = not_found + not_parsed_correctly

errors = {}
errors.update(spelling_variants)
errors.update(mistakes)
errors.update(different_orthographic_conventions)

In [15]:
len(not_found_lemmata)/len(dictionary_entries)

0.268188302425107

In [16]:
len(mistakes)/len(not_found_lemmata)

0.22340425531914893

In [17]:
(len(different_orthographic_conventions)+len(spelling_variants))/len(not_found_lemmata)

0.13297872340425532

In [18]:
len(not_found)/len(not_found_lemmata)

0.010638297872340425

In [19]:
len(compounds)/len(not_found_lemmata)

0.010638297872340425

In [20]:
len(merges)/len(not_found_lemmata)

0.026595744680851064

In [21]:
len(proper_nouns)/len(not_found_lemmata)

0.6648936170212766

There are many reasons that explain why we cannot find 188 annotated lemmata in the Zoëga's dictionary, i.e. around 27% of Völuspá's lemmata:

- here are no proper nouns in the ZD (66%),
- I made mistakes (22%),
- the *ZD* and the *Eddas* were not normalized with the same conventions (13%),
- compound words are not always marked as main entries (2%),
- merged words do not appear clearly in entries (1%),
- some words are not found in the dictionary (1%).

## Proper nouns

* god names
* associated god names
* toponyms
* species names
* creature names
* giant names
* dwarf names
* valkyrie names

## Nature of mistakes

## Orthographic conventions and normalization

##### Letters present in the **Völuspá**:

In [22]:
words = lem_voeluspaa.words()
voeluspaa_characters = sorted(list({c for word in words for c in word if not c.isdigit() and c.islower() and c.isalpha()}))
print(" ".join(voeluspaa_characters))

a b d e f g h i j k l m n o p r s t u v x y z á æ é í ð ó ö ú ý þ œ


Let's keep the lower case characters.

##### Letters present in *ZD*:

In [23]:
dictionary.get_entries()
entries = [entry.word for entry in dictionary.entries if entry is not None]
zd_characters = {c for word in entries if word is not None for c in word if not c.isdigit() and c.islower() and c.isalpha()}

In [24]:
print(" ".join(sorted(list(zd_characters))))

a b d e f g h i j k l m n o p r s t u v w x y z á æ é í ð ó ö ø ú ý þ œ


Charcters w and ø appear only in *ZD*, however the use of w is not significant because there is only one word 'þrywja' with it:

In [25]:
set([word for word in entries if word is not None if "w" in word])

{'þrywja'}

In [26]:
þrywja = dictionary.find("þrywja")
print(þrywja.description.strip())

(þrumda), v. to lie, extend, spread (þrymr of öll lönd ørlögsímu).


*ZD* stated that ǫ, *o ogonoek* is not widely used so the author prefered the modern convention used for Icelandic ö which is a merge of ǫ and ø.

Diplomatica documents of 

 <a href='#gramm_wiki'>The first grammatical treatise</a> is the first Old Norse writing dealing with the best orthographic conventions that Old Norse should have. It can only be found in *Codex Wormianus* with three other grammatical treatises and the Snorri's Edda. See <a href="#traite_theor_ling">[4]</a>

> Vowels: a, ȧ, ǫ, ǫ̇, e, ė, ę, ę̇, ı, i, o, ȯ, ø, ø̇, u, u̇, y, ẏ

> Consonants: b, ʙ, c, ᴋ, d, ᴅ, f, ꜰ, g, ɢ, ǥ, h, l, ʟ, m, ᴍ, n, ɴ, p, ᴘ, r, ʀ, ſ, ꜱ, t, ᴛ, þ

Lower case consonants represent short consonants and upper case consonants are long consonants.

Unfortunately, this orthigraphic reform never happened.


## Comparison with the Menota annotations

(AM 242 fol – Codex Wormianus v. 0.9.9) [http://clarino.uib.no/menota/document-element](http://clarino.uib.no/menota/document-element)

* [Völuspá in Codex Regius](http://clarino.uib.no/menota/texts/GKS-2365-4to-Vsp.xml)
* [Völuspá in Hauksbók](http://clarino.uib.no/menota/texts/AM-544-4to-Vsp.xml)

In [27]:
import os
import codecs
import html

import xml.etree.ElementTree as ET
from xml.sax.saxutils import escape, unescape

In [28]:
print(html.unescape("&et;"))

&et;


In [29]:
konungsbok_filename = "GKS-2365-4to-Vsp.xml"
hausbok_filename = "AM-544-4to-Vsp.xml"
menota_directory = "menota"
#konungsbok = ET.parse(os.path.join(menota_directory, konungsbok_filename))
# hausbok = ET.parse(os.path.join(menota_directory, hausbok_filename))
with codecs.open(os.path.join(menota_directory, konungsbok_filename), "r", encoding="utf-8") as f:
    konungsbok = ET.fromstring(html.unescape(f.read()))
#    ET.parse
#with codecs.open(os.path.join(menota_directory, hausbok_filename), "r", encoding="utf-8") as f:
#     hausbok = ET.fromstring(html.unescape(f.read()))
    
    

ParseError: undefined entity &et;: line 200, column 41 (<string>)

This error is because the namespace of tags is not defined.

In [30]:
from lxml import etree

In [31]:
parser = etree.XMLParser(load_dtd=True, no_network=False)
tree = etree.parse(os.path.join(menota_directory, konungsbok_filename), parser=parser)

In [ ]:
# etree.dump(tree.getroot())

In [32]:
root = tree.getroot()
decoded_root = etree.tostring(root).decode("utf-8")
i = 170
print("\n".join(decoded_root.split("\n")[i:i+10]))

                            <me:dipl>ec</me:dipl>
                        <me:norm>ek</me:norm>
</w>
                        <w me:msa="xAJ rP gF nP cA sI" lemma="allr">
                            <me:facs>allar</me:facs>
                            <me:dipl>allar</me:dipl>
                        <me:norm>allar</me:norm>
</w>
                    </l>
                    <l>


In [33]:
root.tag

'{http://www.tei-c.org/ns/1.0}TEI'

In [34]:
stringify = etree.XPath("string()")

In [35]:
namespaces = {'n': 'http://www.tei-c.org/ns/1.0',
              'me': 'http://www.menota.org/ns/1.0'}

In [36]:
header, text = root.getchildren()

In [37]:
print(text.find('.//n:div[@type="poem"]', namespaces=namespaces))
poem = text.find('.//n:div[@type="poem"]', namespaces=namespaces)

<Element {http://www.tei-c.org/ns/1.0}div at 0x7f4eb1df0688>


In [38]:
poem = text.getchildren()[0].getchildren()[3]

In [39]:
stanzas = [child for child in poem.xpath("n:lg", namespaces=namespaces)]

In [40]:
facsimile_stanzas = [stanza.findall(".//n:l", namespaces=namespaces) for stanza in stanzas]
facsimile_lines = [[line.findall(".//me:facs", namespaces=namespaces) for line in stanza] for stanza in facsimile_stanzas]
facsimile_text = [[[stringify(word) for word in line] for line in stanza ] for stanza in facsimile_lines]

In [41]:
diplomatica_stanzas = [stanza.findall(".//n:l", namespaces=namespaces) for stanza in stanzas]
diplomatica_lines = [[line.findall(".//me:dipl", namespaces=namespaces) for line in stanza] for stanza in diplomatica_stanzas]
diplomatica_text = [[[stringify(word) for word in line] for line in stanza ] for stanza in diplomatica_lines]

# diplomatica = [line.findall(".//me:dipl", namespaces=namespaces) for line in lines]
# diplomatica_text = [[stringify(word) for word in line] for line in diplomatica]

In [42]:
# normalized = [ for line in lines]
facsimile_text

[[['Hlıoðſ', 'bið', 'ec', 'allar'],
  ['kinꝺir'],
  ['meiri', '⁊', 'miɴi'],
  ['maugo', 'heimꝺalar'],
  ['uilðo', 'at', 'ec', 'ualꝼꜹþ̅'],
  ['uel', 'ꝼyr', 'telia'],
  ['ꝼoꝛn', 'ſpioll', 'ꝼíra'],
  ['þꜹ', 'e͛', 'ꝼremſt', 'u̅', 'man', '.']],
 [['Ec', 'mán', 'iǫtna'],
  ['ar', 'um', 'boꝛna'],
  ['þꜹ̣', 'e͛', 'ꝼoꝛꝺo̅', 'mic'],
  ['ꝼǫꝺꝺa', 'hoꝼꝺo', '.'],
  ['nio', 'man', 'ec', 'hei̅a'],
  ['nío', 'iviþi[᷑]'],
  ['miot uið', 'm\ue499ran'],
  ['ꝼͮ', 'molꝺ', 'neðan', '.']],
 [['Ar', 'uar', 'alꝺa'],
  ['þar', 'er', 'ẏm͛', 'bygði'],
  ['vara', 'ſanꝺꝛ', 'nę', 'ſ\ue499r'],
  ['ne', 'ſualar', 'uɴ͛'],
  ['ioꝛð', 'ꝼaɴz', 'ęva'],
  ['ne', 'uṗ him̅'],
  ['gap', 'uar', 'gꝛiɴvnga'],
  ['eɴ', 'graſ', 'hv͛gi', '.']],
 [['Aꝺꝛ', 'bvrſ', 'ſẏn͛'],
  ['bioðo̅', 'v̅', 'ẏpðo'],
  ['þ̅r', 'er', 'miðgarꝺ'],
  ['męran', 'ſcópo', '.'],
  ['ſól', 'ſceín', 'ſuɴan'],
  ['aꝼ̣', 'ſalar', 'ſteina'],
  ['þa', 'var', 'grvnꝺ', 'groin'],
  ['grǫno̅', 'lauki', '.']],
 [['Sol', 'varp', 'ſvɴan'],
  ['ſiɴi', 'mana'],
  ['henꝺi', 

In [43]:
diplomatica_text

[[['Hlıoðſ', 'bið', 'ec', 'allar'],
  ['kinꝺir'],
  ['meiri', 'oc', 'miɴi'],
  ['maugo', 'heimꝺalar'],
  ['uilðo', 'at', 'ec', 'ualꝼꜹþr'],
  ['uel', 'ꝼyr', 'telia'],
  ['ꝼoꝛn', 'ſpioll', 'ꝼíra'],
  ['þꜹ', 'er', 'ꝼremſt', 'um', 'man', '.']],
 [['Ec', 'mán', 'iǫtna'],
  ['ar', 'um', 'boꝛna'],
  ['þꜹ̣a', 'er', 'ꝼoꝛꝺom', 'mic'],
  ['ꝼǫꝺꝺa', 'hoꝼꝺo', '.'],
  ['nio', 'man', 'ec', 'heima'],
  ['nío', 'iviþi[vr]'],
  ['miot uið', 'm\ue499ran'],
  ['ꝼyr', 'molꝺ', 'neðan', '.']],
 [['Ar', 'uar', 'alꝺa'],
  ['þar', 'er', 'ẏmir', 'bygði'],
  ['vara', 'ſanꝺꝛ', 'nę', 'ſ\ue499r'],
  ['ne', 'ſualar', 'uɴir'],
  ['ioꝛð', 'ꝼaɴz', 'ęva'],
  ['ne', 'upp  himiɴ'],
  ['gap', 'uar', 'gꝛiɴvnga'],
  ['eɴ', 'graſ', 'hvergi', '.']],
 [['Aꝺꝛ', 'bvrſ', 'ſẏnir'],
  ['bioðom', 'vm', 'ẏpðo'],
  ['þeir', 'er', 'miðgarꝺ'],
  ['męran', 'ſcópo', '.'],
  ['ſól', 'ſceín', 'ſuɴan'],
  ['aꝼ̣', 'ſalar', 'ſteina'],
  ['þa', 'var', 'grvnꝺ', 'groin'],
  ['grǫnom', 'lauki', '.']],
 [['Sol', 'varp', 'ſvɴan'],
  ['ſiɴi', 'mana'],
 

In [44]:
normalized_stanzas = [stanza.findall(".//n:l", namespaces=namespaces) for stanza in stanzas]
normalized_lines = [[line.findall(".//me:norm", namespaces=namespaces) for line in stanza] for stanza in normalized_stanzas]
normalized_text = [[[stringify(word) for word in line] for line in stanza ] for stanza in normalized_lines]

In [45]:
normalized_text

[[['Hljóðs', 'bið', 'ek', 'allar'],
  ['kindir'],
  ['meiri', 'ok', 'minni'],
  ['mögu', 'Heimdalar'],
  ['vildu', 'að', 'ek', 'Valföðr'],
  ['vel', 'fyr', 'telja'],
  ['forn', 'spjöll', 'fira'],
  ['þau', 'er', 'fremst', 'um', 'man', '.']],
 [['Ek', 'man', 'jötna'],
  ['ár', 'um', 'borna'],
  ['þá', 'er', 'forðum', 'mik'],
  ['fædda', 'höfðu', '.'],
  ['Níu', 'man', 'ek', 'heima'],
  ['níu', 'íviðjur'],
  ['mjötvið', 'mæran'],
  ['fyr', 'mold', 'neðan', '.']],
 [['Ár', 'var', 'alda'],
  ['þar', 'er', 'Ýmis', 'byggði'],
  ['var-a', 'sandr', 'né', 'sær'],
  ['né', 'svalar', 'unnir'],
  ['jörð', 'fannz', 'æva'],
  ['ne', 'upphiminn'],
  ['gap', 'var'],
  ['en', 'gras', 'hvergi', '.']],
 [['Áðr', 'Burs', 'synir'],
  ['bjöðum', 'um', 'yppðu'],
  ['þeir', 'er', 'Miðgarð'],
  ['mæran', 'skópu', '.'],
  ['Sól', 'skein', 'sunnan'],
  ['á', 'salar', 'steina'],
  ['þá', 'var', 'grund', 'gróin'],
  ['grænum', 'lauki', '.']],
 [['Sól', 'varp', 'sunnan'],
  ['sinni', 'mána'],
  ['hendi', 'inni', 'h

In [55]:
lemmata_stanzas = [stanza.findall(".//n:l", namespaces=namespaces) for stanza in stanzas]
lemmata_lines = [[line.findall(".//n:w", namespaces=namespaces) for line in stanza] for stanza in lemmata_stanzas]
lemmata_text = [[[word.get("lemma") for word in line] for line in stanza ] for stanza in lemmata_lines]

In [56]:
lemmata_text

[[['hlióð', 'biðia', 'ek', 'allr'],
  ['kind'],
  ['mikill', 'ok', 'lítill'],
  ['mǫgr', 'Heimdallr'],
  ['vilia', 'at', 'ek', 'Valfǫðr'],
  ['vel', 'fyr', 'telia'],
  ['forn', 'spiall', 'firar'],
  ['sá', 'er', 'framr', 'um', 'muna']],
 [['ek', 'muna', 'iǫtunn'],
  ['ár', 'um', 'bera'],
  ['sá', 'er', 'forðum', 'ek'],
  ['fǿða', 'hafa'],
  ['níu', 'muna', 'ek', 'heimr'],
  ['níu', None],
  ['mjǫtviðr', 'mǽrr'],
  ['fyr', 'mold', 'neðan']],
 [['ár', 'vera', 'ǫld'],
  ['þar', 'er', 'Ýmir', 'byggia'],
  ['vera', 'sandr', 'ne', 'sǽr'],
  ['ne', 'svalr', 'unnr'],
  ['iǫrð', 'finna', 'æva'],
  ['ne', 'upphiminn'],
  ['gap', 'vera', None],
  ['en', 'gras', 'hvergi']],
 [['áðr', 'Burr', 'sonr'],
  ['biǫð', 'um', 'yppa'],
  ['sá', 'er', 'Miðgarðr'],
  ['mǽrr', 'skapa'],
  ['sól', 'skína', 'sunnan'],
  ['á', 'salr', 'steinn'],
  ['þá', 'vera', 'grund', 'gróa'],
  ['grǿnn', 'laukr']],
 [['sól', 'verpa', 'sunnan'],
  ['sinni', 'máni'],
  ['hǫnd', 'inn', 'hǿgri'],
  ['um', 'himinn', 'iódýr'],
  ['

## Sources:

1. <a id='the_destination'></a>

2. <a id='gramm_wiki'></a>[The first grammatical treatise](https://en.wikipedia.org/wiki/First_Grammatical_Treatise)
3. Normalized text of the first grammatical treatise [http://etext.old.no/gramm/](http://etext.old.no/gramm/)
4. <a id='traite_theor_ling'></a>[Le Premier traité grammatical islandais dans l’histoire des théories linguistiques](http://ctlf.ens-lyon.fr/documents/articles/Ptgi.Mikkelsen.Guelpa.pdf) in French
5. [Old Norse normalized texts](https://heimskringla.no/wiki/Norr%C3%B8ne_kildetekster)
6. [Pictures codex Wormianus](http://www.e-pages.dk/ku/621/) p.84 -> First Grammatical Treatise
7. [Handbok i norrøne filologi](https://folk.uib.no/hnooh/handbok/)

**Work** done with the following versions:
- [https://github.com/cltk/cltk](): 0.1.110
- https://github.com/cltk/old_norse_texts_heimskringla/: 1.2.0 
- https://github.com/cltk/old_norse_dictionary_zoega: 1.0.6

